uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/AbsencePeriod.php

```diff
@@ -308,28 +310,127 @@ public function getNumberOfWorkingDays()
    *
    * @return int
    */
-  public function getNumberOfWorkingDaysToWork($startDate, $endDate)
-  {
-    if(!CRM_HRLeaveAndAbsences_Validator_Date::isValid($startDate, 'Y-m-d')) {
+  public function getNumberOfWorkingDaysToWork($startDate, $endDate) {
+    if (!CRM_HRLeaveAndAbsences_Validator_Date::isValid($startDate, 'Y-m-d')) {
       throw new InvalidArgumentException('getNumberOfWorkingDaysToWork expects a valid startDate in Y-m-d format');
     }
 
-    if(!CRM_HRLeaveAndAbsences_Validator_Date::isValid($endDate, 'Y-m-d')) {
+    if (!CRM_HRLeaveAndAbsences_Validator_Date::isValid($endDate, 'Y-m-d')) {
       throw new InvalidArgumentException('getNumberOfWorkingDaysToWork expects a valid endDate in Y-m-d format');
     }
 
-    if(strtotime($startDate) < strtotime($this->start_date)) {
+    if (strtotime($startDate) < strtotime($this->start_date)) {
       $startDate = $this->start_date;
     }
 
-    if(strtotime($endDate) > strtotime($this->end_date)) {
+    if (strtotime($endDate) > strtotime($this->end_date)) {
       $endDate = $this->end_date;
     }
 
-    $periodToWork = new CRM_HRLeaveAndAbsences_BAO_AbsencePeriod();
+    $periodToWork             = new CRM_HRLeaveAndAbsences_BAO_AbsencePeriod();
     $periodToWork->start_date = $startDate;
-    $periodToWork->end_date = $endDate;
+    $periodToWork->end_date   = $endDate;
 
     return $periodToWork->getNumberOfWorkingDays();
   }
+
+  /**
+   * Returns the Absence Period previous to this one. That is, the Absence
+   * Period where weight is equal to this Period weight - 1.
+   *
+   * @return null|CRM_HRLeaveAndAbsences_BAO_AbsencePeriod - The previous Absence Period or null if there's none
+   */
+  public function getPreviousPeriod()
+  {
+    $previousPeriod = new CRM_HRLeaveAndAbsences_BAO_AbsencePeriod();
```

Comment:
same here 

> why not to change this to :

new $className;

Also I think there is a php function to retrieve the current class name .. maybe its better to use it ?!! what do you think ?